clap_complete/src/dynamic/completer.rs

```diff
@@ -76,8 +76,90 @@ pub fn complete(
+                                ParseState::ValueDone
+                            } else {
+                                ParseState::Opt(o.clone())
+                            }
+                        }
+                        clap::ArgAction::SetTrue | clap::ArgAction::SetFalse => {
+                            ParseState::ValueDone
+                        }
+                        clap::ArgAction::Count => ParseState::ValueDone,
+                        clap::ArgAction::Version => ParseState::ValueDone,
+                        clap::ArgAction::Help
+                        | clap::ArgAction::HelpLong
+                        | clap::ArgAction::HelpShort => ParseState::ValueDone,
+                        _ => ParseState::ValueDone,
+                    }
+                } else {
+                    ParseState::ValueDone
+                };
+            } else {
+                state = ParseState::ValueDone;
+            }
+        } else if let Some(short) = arg.to_short() {
+            let mut short = short.clone();
+
+            loop {
+                if let Some(Ok(opt)) = short.next_flag() {
+                    let opt = current_cmd.get_arguments().find(|a| {
+                        let shorts = a.get_short_and_visible_aliases();
+                        let is_find = shorts.map(|v| {
+                            let mut iter = v.into_iter();
+                            let c = iter.find(|c| *c == opt);
+                            c.is_some()
+                        });
+                        is_find.unwrap_or(false)
+                    });
+                    let mut flag = false;
+                    state = if let Some(o) = opt {
+                        match o.get_action() {
+                            clap::ArgAction::Set | clap::ArgAction::Append => {
+                                flag = true;
+                                if short.next_value_os().is_some() {
+                                    ParseState::ValueDone
+                                } else {
+                                    ParseState::Opt(o.clone())
+                                }
+                            }
+                            clap::ArgAction::SetTrue | clap::ArgAction::SetFalse => {
+                                ParseState::ValueDone
+                            }
+                            clap::ArgAction::Count => ParseState::ValueDone,
+                            clap::ArgAction::Version => ParseState::ValueDone,
+                            clap::ArgAction::Help
+                            | clap::ArgAction::HelpShort
+                            | clap::ArgAction::HelpLong => ParseState::ValueDone,
+                            _ => ParseState::ValueDone,
+                        }
+                    } else {
+                        ParseState::ValueDone
+                    };
+
+                    if flag {
+                        break;
+                    }
```

Comment:
`flag` is confusing because they are all flags.  I think you mean `takes_value`?